flink-core/src/main/java/org/apache/flink/api/common/state/v2/ReducingStateDescriptor.java

```diff
@@ -67,6 +67,23 @@ public ReducingStateDescriptor(
         this.reduceFunction = checkNotNull(reduceFunction);
     }
 
+    /**
+     * Creates a new {@code ReducingStateDescriptor} with the given name, type, and default value.
+     *
+     * <p>If this constructor fails (because it is not possible to describe the type via a class),
+     * consider using the {@link #ReducingStateDescriptor(String, ReduceFunction, TypeInformation)}
+     * constructor.
```

Comment:
it would be good to detail why this advice is being given.

Reply:
There is no difference for these constructors. If this fails, then use another, isn't this clear?